src/hotspot/share/opto/compile.cpp

```diff
@@ -5094,6 +5107,15 @@ void Compile::print_method(CompilerPhaseType cpt, int level, Node* n) {
   C->_latest_stage_start_counter.stamp();
 }
 
+void Compile::print_method_iter(CompilerPhaseType cpt, int level, Node* n) {
+#ifndef PRODUCT
+  int iter = ++_igv_phase_iter[cpt];
+  print_method(cpt, level, n, iter);
+#else
+  print_method(cpt, level, n);
```

Comment:
This is dead code because all calls are guarded by `NOT_PRODUCT`, right?

Reply:
Not quite, there is one unguarded call for `PHASE_PHASEIDEALLOOP_ITERATIONS` at level 2. This is an existing phase that I converted from `print_method` to `print_method_iter` (as suggested by @chhagedorn in the issue corresponding to this PR).

Reply:
I have now incorporated the functionality of `print_method_iter` directly into `print_method`, and also removed all the `NOT_PRODUCT` wrappers.

I'm resolving this thread now, should we move the discussion regarding JFR event dumping somewhere else?

Reply:
Maybe we can mention it as a side node in the PR description that it is something we should think about at some point. But it should not block this PR.

Reply:
Sure, I'll add it to the final (non-draft) PR description.